### PR TITLE
raft: raft learners should be returned after applyConfChange

### PR DESCRIPTION
**Pull request by absolute8511**
This should fix issue #9087

### CODE REVIEW

**Comment by siddontang** [comment]
Do we use the node function outside of raft lib? if yes, we should also handle learner nodes now.

**Comment by absolute8511** [comment]
Currently, as I know it is used only by raft lib. If any future usage maybe another pr needed.

File: raft/node_test.go [hunk 1]
```diff
@@ -728,3 +728,54 @@ func TestIsHardStateEqual(t *testing.T) {
...
+			select {
+			case <-stop:
+				return
+			case <-ticker.C:
+				n.Tick()
+			case rd := <-n.Ready():
+				s.Append(rd.Entries)
+				t.Logf("raft: %v", rd.Entries)
+				for _, ent := range rd.Entries {
+					if ent.Type == raftpb.EntryConfChange {
```
Comment (xiang90) [inline raft/node_test.go]: indentation is crazy here. let us kill one indentation by doing 

```
if ent.Type != raftpb.EntryConfChange {
    continue
}
var cc raftpb.ConfChange
...
```

File: raft/node_test.go [hunk 2]
```diff
@@ -728,3 +728,54 @@ func TestIsHardStateEqual(t *testing.T) {
...
+						cc.Unmarshal(ent.Data)
+						state := n.ApplyConfChange(cc)
+						if len(state.Learners) == 0 ||
+							state.Learners[0] != cc.NodeID ||
+							cc.NodeID != 2 {
+							t.Fatalf("apply conf change should return new added learner: %v", state.String())
+						}
+
+						if len(state.Nodes) != 1 {
+							t.Fatalf("add learner should not change the nodes: %v", state.String())
```
Comment (xiang90) [inline raft/node_test.go]: use t.errorf

**Comment by xiang90** [comment]
@absolute8511 Can you rebase with master?

Looks good to me. @siddontang can you give this a final look?

File: raft/node_test.go (hunk 1 above)
Comment (siddontang) [reply raft/node_test.go]: +1

**Comment by siddontang** [comment]
Rest LGTM

**Comment by xiang90** [comment]
@absolute8511 can you please rebase with master, and resolve the merge conflicts? thank you.

**Comment by absolute8511** [comment]
All done @xiang90

**Pull request merged**